module/main.go

```diff
@@ -12,11 +14,42 @@ import (
 	viamcartographer "github.com/viamrobotics/viam-cartographer"
 )
 
+// Versioning variables which are replaced by LD flags.
+var (
+	Version     = "development"
+	GitRevision = ""
+)
+
 func main() {
 	utils.ContextualMain(mainWithArgs, golog.NewLogger("cartographerModule"))
 }
 
 func mainWithArgs(ctx context.Context, args []string, logger golog.Logger) error {
+	var versionFields []interface{}
+	// print version on statup
+	// exit 0 if --version or -version argv is provided
+	// NOTE: utils.ParseFlags can't be used w/o a nolint as it
+	// errors if there are args not described by the struct,
+	// which is the case during normal module use.
+	if len(args) != 2 {
```

Comment:
Why is this length 2? If you run `carto-module --version` isn't that just 1 arg?

Reply:
In linux/unix, ARGV[0] is the name of the command itself. Go does this is well with os.Args (which is where this ultimately comes from.